cpp/email_watcher.cc

```diff
@@ -198,21 +189,62 @@ std::map<std::string, time_t> LoadSectionNamesToLastSeenTimeWindowsMap() {
     return section_names_to_last_seen_time_map;
 }
 
-
+#if 0
 void SaveSectionNamesToLastSeenTimeWindowsMap(const std::map<std::string, time_t> &section_names_to_last_seen_time_map) {
     const auto output(FileUtil::OpenOutputFileOrDie(MAPFILE_PATH));
     for (const auto &[section_name, last_seen_time] : section_names_to_last_seen_time_map)
         (*output) << section_name << '=' << last_seen_time << '\n';
     LOG_INFO("Wrote " + std::to_string(section_names_to_last_seen_time_map.size()) + " entries/entry to " + MAPFILE_PATH + ".");
 }
+#endif
+
+void SendNotificationsForOverdueEmails(const IniFile &ini_file, std::set<std::string> * const matched_section_names,
+                                       const std::vector<std::string> &notification_email_addresses, const time_t NOW,
+                                       const std::map<std::string, time_t> &section_names_to_last_seen_time_map)
+{
+    std::string overdue_list;
+    for (const auto &section : ini_file) {
+        const auto &section_name(section.getSectionName());
+        if (section_name.empty()) // Global section
+            continue;
+
+        LOG_DEBUG("Processing section " + section_name + ".");
+
+        if (matched_section_names->find(section_name) != matched_section_names->end()) {
+            LOG_DEBUG("\tWe have new mail for " + section_name + " and therefore can't be overdue!");
+            continue; // We're definitely *not* overdue!
+        }
+
+        const auto section_name_and_last_seen_time(section_names_to_last_seen_time_map.find(section_name));
+        if (section_name_and_last_seen_time == section_names_to_last_seen_time_map.end()) {
+            LOG_DEBUG("\tSection " + section_name + " not found in section_names_to_last_seen_time_map!");
+            matched_section_names->emplace(section_name);
+            continue;
+        }
+        LOG_DEBUG("\tLast seen " + TimeUtil::TimeTToString(section_name_and_last_seen_time->second) + ".");
+
+        const unsigned overdue_time_window(section.getUnsigned("overdue_time_window")); // in hours
+        if ((NOW - section_name_and_last_seen_time->second) > overdue_time_window * 3600u) {
+            overdue_list += "No email from " + section.getString("from_host") + " with subject pattern "
+                            + section.getString("subject_pattern") + " within the last "
```

Comment:
subject_pattern should be quoted somehow

Reply:
Done!